tf-modules/data-persistence/elasticsearch.tf

```diff
@@ -0,0 +1,118 @@
+locals {
+  inside_vpc         = length(var.subnet_ids) > 0 ? true : false
+  deploy_inside_vpc  = var.include_elasticsearch && local.inside_vpc
+  deploy_outside_vpc = var.include_elasticsearch && local.inside_vpc == false
+  include_es_policy  = length(var.es_trusted_role_arns) > 0 ? true : false
+  es_domain_name     = "${var.prefix}-${var.elasticsearch_config.domain_name}${local.inside_vpc ? "-vpc" : ""}"
+}
+
+data "aws_region" "current" {}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "es_access_policy" {
+  statement {
+    actions = [
+      "es:*"
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers =  distinct(compact(var.es_trusted_role_arns))
+    }
+
+    resources = [
+      "arn:aws:es:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:domain/${local.es_domain_name}/*"
```

Comment:
```suggestion
      "${aws_elasticsearch_domain.es.arn}/*"
```

Reply:
Won't this depend on whether `es` or `es_vpc` is the domain being used?

Reply:
In that case,

```hcl
resources = compact(
      [
        local.deploy_inside_vpc ? "${aws_elasticsearch_domain.es_vpc.arn}/*"
        local.deploy_outside_vpc ? "${aws_elasticsearch_domain.es.arn}/*"
      ]
    )
```